Formula/rat.rb

```diff
@@ -0,0 +1,26 @@
+class Rat < Formula
+  desc "Compose shell commands to build terminal applications"
+  homepage "https://github.com/ericfreese/rat"
+  url "https://github.com/ericfreese/rat.git"
+  version "0.0.2"
+  sha256 "9da8d67a1e16056e3f05cdf536766e5571ede54a6c241c941d6c74eee5d64d31"
+
+  depends_on "go" => :build
+
+  def install
+    contents = Dir["{*,.git,.gitignore}"]
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/ericfreese/rat").install contents
+
+    ENV["GOPATH"] = gopath
+
+    cd gopath/"src/github.com/ericfreese/rat" do
+      system "go", "build", "-o", "rat"
+      bin.install "rat"
```

Comment:
Using this format, you'll need a `prefix.install_metafiles` after this line, I suspect.

Reply:
Not sure what that does, but I based this roughly on [`consul`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/consul.rb) and I don't see `prefix.install_metafiles` in there.

Reply:
Yes, `consul` needs the line too, heh. Good spot.

Reply:
Can you give me an overview on what that does and why I need it or point me to some documentation?

Is it a blocker to merging this PR?

Reply:
Kinda does what it says on the tin. Install the metafiles: `README,` `LICENSE`, etc. Homebrew usually handles this automatically but if you transpose the contents of the buildpath, which you are doing here, that automatic logic can break down. It's a known issue.

Reply:
Added:

```diff
diff --git a/Formula/rat.rb b/Formula/rat.rb
index b286652d46..66dd37dcee 100644
--- a/Formula/rat.rb
+++ b/Formula/rat.rb
@@ -16,6 +16,7 @@ class Rat < Formula
     cd gopath/"src/github.com/ericfreese/rat" do
       system "go", "build", "-o", "rat"
       bin.install "rat"
+      prefix.install_metafiles
     end
   end
```